### PR TITLE
Adpat timeout value of torch.dist.init_process_group depending on elapsed time

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary
When huge dataset is used, more than 30 seconds is used to prepare dataset, which raises an error when training with multi GPU because current `torch.dist.init_process_group` timeout is set as 30.
So, I implemented a feature to adapt timeout value if process start time is given.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ All notable changes to this project will be documented in this file.
 
 ### Enhancements
 
--
+- Adapt timeout value of initialization for distributed training(<https://github.com/openvinotoolkit/training_extensions/pull/2422>)
```
Comment (goodsong81) [inline CHANGELOG.md]:
```suggestion
- Adapt timeout value of initialization for distributed training (<https://github.com/openvinotoolkit/training_extensions/pull/2422>)
```

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Sorry for the nitpick but could you fix the changelog while resoling the conflict anyway? :)

**Comment by sungmanc** [comment]
I have a question, are `preparing the dataset` and `dist group initialization` conducted in parallel? why `prepare dataset` affects the `dist group initialization`?

**Review by goodsong81 — DISMISSED** [review]

**Comment by eunwoosh** [comment]
> I have a question, are `preparing the dataset` and `dist group initialization` conducted in parallel? why `prepare dataset` affects the `dist group initialization`?

Spawning multi process is executed after preparing dataset. So, When main process initializes first and waits for child process initialization, child process should prepare dataset first before initialization. So if dataset preparation is more than timeout seconds, timeout error is raised.

**Comment by sungmanc** [comment]
> > I have a question, are `preparing the dataset` and `dist group initialization` conducted in parallel? why `prepare dataset` affects the `dist group initialization`?
> 
> Spawning multi process is executed after preparing dataset. So, When main process initializes first and waits for child process initialization, child process should prepare dataset first before initialization. So if dataset preparation is more than timeout seconds, timeout error is raised.

Understood, thanks for the kind explanation

**Review by sungmanc — DISMISSED** [review]

**Comment by eunwoosh** [comment]
Sorry I add a commit to resolve conflict of CHANGELOG.md. Could you review PR again, @sungmanc @goodsong81

**Review by sungmanc — APPROVED** [review]
No problem :)

**Review by goodsong81 — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Pull request merged**